tests/infra_test.go

```diff
@@ -448,41 +418,32 @@ func takeMetricsWithPrefix(output, prefix string) []string {
 	return ret
 }
 
-// shortcut helper
-func newRandomVMIWithMetrics(virtClient kubecli.KubevirtClient, metricSubstring string) ([]*v1.VirtualMachineInstance, string, map[string]float64) {
-	return newRandomVMIsWithMetrics(1, "", virtClient, metricSubstring)
-}
-
-func newRandomVMIsWithMetrics(count int, preferredNodeName string, virtClient kubecli.KubevirtClient, metricSubstring string) ([]*v1.VirtualMachineInstance, string, map[string]float64) {
+func newRandomVMIWithMetrics(virtClient kubecli.KubevirtClient, preferredNodeName, metricSubstring string) ([]*v1.VirtualMachineInstance, string, map[string]float64) {
```

Comment:
the simplification looks OK, but I think this utility function can be kept as is, should we need to add more than a single VMI to the future test cases.

Reply:
Seems like you looked at a middle step in this PR, right?
What bothers me in this function is that it does two things - run the VMI and gather its metrics. The former can be used for running the first VM and the latter seems to be redundant at this point (as we intentionally target 2 VMs due to resource constraints in the CI, right?)
Maybe we can discuss the final step, where this function is removed, and think if it makes sense or not

Reply:
Yes, I was reviewing a commit a time in chronological order. And yes, I rolled two responsabilities in one function. It's probably better to split them.